certbot/tests/log_test.py

```diff
@@ -27,12 +26,33 @@ def _call(cls, *args, **kwargs):
         return pre_arg_parse_setup(*args, **kwargs)
 
     def test_it(self):
-        with mock.patch('certbot.log.except_hook') as mock_except_hook:
-            with mock.patch('certbot.log.sys') as mock_sys:
-                self._call()
-
+        with mock.patch('certbot.log.util.atexit_register') as mock_register:
```

Comment:
I knew it was a thing for `if`s, but I thought the whole point of the `with` was to clean things up at the end, making this scope pattern rather absurd...